manager/controllers/advisory_detail.go

```diff
@@ -16,48 +16,92 @@ import (
 
 var enableAdvisoryDetailCache = utils.GetBoolEnvOrDefault("ENABLE_ADVISORY_DETAIL_CACHE", true)
 var advisoryDetailCacheSize = utils.GetIntEnvOrDefault("ADVISORY_DETAIL_CACHE_SIZE", 100)
-var advisoryDetailCache = initAdvisoryDetailCache()
+var advisoryDetailCacheV1 = initAdvisoryDetailCache()
+var advisoryDetailCacheV2 = initAdvisoryDetailCache()
 
-type AdvisoryDetailResponse struct {
-	Data AdvisoryDetailItem `json:"data"`
+type AdvisoryDetailResponseV1 struct {
+	Data AdvisoryDetailItemV1 `json:"data"`
 }
 
-type AdvisoryDetailItem struct {
-	Attributes AdvisoryDetailAttributes `json:"attributes"`
-	ID         string                   `json:"id"`
-	Type       string                   `json:"type"`
+type AdvisoryDetailResponseV2 struct {
+	Data AdvisoryDetailItemV2 `json:"data"`
+}
+
+type AdvisoryDetailItemV1 struct {
+	Attributes AdvisoryDetailAttributesV1 `json:"attributes"`
+	ID         string                     `json:"id"`
+	Type       string                     `json:"type"`
+}
+
+type AdvisoryDetailItemV2 struct {
+	Attributes AdvisoryDetailAttributesV2 `json:"attributes"`
+	ID         string                     `json:"id"`
+	Type       string                     `json:"type"`
```

Comment:
Nitpick: you can put shared attributes in common type
```
type AdvisoryDetailItem struct {
	ID         string                     `json:"id"`
	Type       string                     `json:"type"`
}

type AdvisoryDetailItemV1 struct {
        AdvisoryDetailItem
	Attributes AdvisoryDetailAttributesV1 `json:"attributes"`
}

type AdvisoryDetailItemV2 struct {
        AdvisoryDetailItem
	Attributes AdvisoryDetailAttributesV2 `json:"attributes"`
}
```

Reply:
thanks